storage/metricsstore/prometheus/tests/test_helper.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/storage/metricsstore"
+	"github.com/jaegertracing/jaeger/storage/metricsstore/prometheus"
+)
+
+type (
+	newMetricsReaderFn func(*zap.Logger, []string, time.Duration) (prometheus.Reader, error)
+)
+
+// TestAll is a convenience function for Prometheus-compliant metrics storage implementations
+// to execute all "common" Prometheus tests to avoid duplicating calls to individual test functions.
+func TestAll(t *testing.T, newMetricsReader newMetricsReaderFn, metricsStoreType string) {
+	TestNewMetricsReader(t, newMetricsReader)
+	TestGetLatencies(t, newMetricsReader)
+	TestGetCallRates(t, newMetricsReader)
+	TestGetErrorRates(t, newMetricsReader)
+	TestGetMinStepDuration(t, newMetricsReader)
+}
+
+func TestNewMetricsReader(t *testing.T, newMetricsReader newMetricsReaderFn) {
```

Comment:
These tests will inevitably be changed and are here just to pass line coverage while there is no Prometheus implementation.